src/metabase/api/util.clj

```diff
@@ -42,4 +44,17 @@
   {:system-info (troubleshooting/system-info)
    :metabase-info (troubleshooting/metabase-info)})
 
+; FIXME: only enable for test + sanitize `name`
+(api/defendpoint POST "/snapshot/:name"
+  [name]
+  (jdbc/query (db/connection) ["SCRIPT TO ?", name])
+  nil)
+
+; FIXME: only enable for test + sanitize `name`
+(api/defendpoint POST "/restore/:name"
+  [name]
+  (jdbc/execute! (db/connection) ["DROP ALL OBJECTS"])
+  (jdbc/execute! (db/connection) ["RUNSCRIPT FROM ?", name])
+  nil)
+
```

Comment:
@flyingmachine @walterl what's the best/safest way to enable these only in tests or with a env var?

I think once that's locked down, we can start using this!

Reply:
AFAIK we don't currently have such a flag/indicator in the back-end, but I vaguely recall that this has been discussed before.

How is the back-end started when this code is used? From a `lein` command, `yarn`, or something else?

Reply:
We use an uberjar in e2e tests both locally and on CI.

Reply:
@sbelak @flyingmachine What would you think about a `(defsetting execution-env...)`?

Reply:
`metabase.config/is-test?`?

Reply:
@camsaul What do you think?

Reply:
Since the tests run against the uberjar I think it would be better to add a separate env var to enable these endpoints conditionally. Something like `ENABLE_TEST_ENDPOINTS` would make sense IMO. You don't want to set `MB_RUN_MODE` to `test` or it's going to do stuff like disable background tasks that you're not going to want.

I would put these endpoints in a separate `testing` namespace e.g. `metabase.api.testing` and only include `/testing` in the API `routes` if the env var is truthy.

Reply:
Basically something like 

```clj
;; metabase.api.routes namespace 

(require '[metabase.api.testing as api.testing])

(def routes
  ...
  (context "/testing" [] (if config/enable-test-endpoints? api.testing/routes (constantly nil))))
```

The basic idea is if test endpoints are enabled (`ENABLE_TEST_ENDPOINTS=true`) it will handle `/api/testing` requests otherwise the `context` handler will return `nil` (in effect saying it can't handle the route) so it will ultimately return a 404